tests/lib/cpp/cxx/testcase.yaml

```diff
@@ -25,3 +25,33 @@ tests:
     filter: CONFIG_PICOLIBC_SUPPORTED
     extra_configs:
       - CONFIG_PICOLIBC=y
+  cpp.main.cpp98:
+    build_only: true
+    extra_configs:
+      - CONFIG_STD_CPP98=y
+  cpp.main.cpp11:
+    build_only: true
+    extra_configs:
+      - CONFIG_STD_CPP11=y
```

Comment:
Only until the default changes, though.  If it does, and I apply this, then "cpp.main.cpp11" will be testing some standard other than C++11!  I think that's clearly not what we want; it's not illegal to specify a kconfig in two places.

Reply:
What I meant was to drop `cpp.main.cpp11` altogether. We do not need an additional testcase that tests the same thing that other testcases above test.

Maybe it would be helpful to add a comment saying "C++11 case is not explicitly tested because it is the default option and already tested by the above testcases."

Reply:
Fair enough